upstairs/src/lib.rs

```diff
@@ -1909,17 +2033,38 @@ impl EncryptionContext {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Serialize)]
+enum UpState {
+    /*
+     * The upstairs is just coming online.  We can send IO on behalf of
+     * the upstairs, but no IO from the guest.
+     */
+    Initializing,
+    /*
+     * The upstairs is online and accepting IO from the guest.
+     */
+    Active,
+    /*
+     * Let in flight IO continue, but don't allow any new IO.  This state
+     * also means that when a downstairs task has completed all the IO
+     * it can, including the final flush, it should reset itself back to
+     * new and let the connection to the downstairs close and let the
+     * loop to reconnect (looper) happen.
```

Comment:
Do we need a state in between when a Guest called deactivate, and when a Guest calls activate? Immediately retrying to reconnect seems problematic here.

Reply:
The upstairs will stay in state `Deactivating` until all downstairs have process the deactivation.
During this time, no new commands will be accepted from the Guest (including an activation).
Once all downstairs have completed deactivation, the upstairs will transition back to `Initializing`, which is where the upstairs will be listening for an activation as well as moving downstairs through their negotiation and, if required, reconciliation.

The downstairs themselves will reflect where they are in the path to activation.

Reply:
In the migration case, Upstairs 1 will deactivate, which puts it into `Initializing`. This means it will negotate with the downstairs but not activate. Say they are both active at the same time. Upstairs 2 will also negotiate, and then activate. I don't think there's a conflict here like I thought there was. Resolving this.